nekoyume/Assets/_Scripts/UI/Rank.cs

```diff
@@ -263,14 +261,125 @@ private async void UpdateCategoryAsync(RankCategory category, bool toggleOn)
                         myInfoCell.SetEmpty(states.CurrentAvatarState);
                     }
 
-                    rankScroll.Show(mimisbrunnrRankingInfos, true);
+                    rankScroll.Show(SharedModel.MimisbrunnrRankingInfos, true);
+                    break;
+                case RankCategory.Crafting:
+                    if (!isApiLoaded)
+                    {
+                        break;
+                    }
+
+                    if (SharedModel.AgentCraftRankingInfos
+                        .TryGetValue(states.CurrentAvatarKey, out var craftingInfo))
+                    {
+                        myInfoCell.SetDataAsCrafting(craftingInfo);
+                    }
+                    else
+                    {
+                        myInfoCell.SetEmpty(states.CurrentAvatarState);
+                    }
+
+                    rankScroll.Show(SharedModel.CraftRankingInfos, true);
+                    break;
+                case RankCategory.EquipmentWeapon:
+                    if (!isApiLoaded)
+                    {
+                        break;
+                    }
+
+                    var weaponRankingInfos = SharedModel.EquipmentRankingInfosMap[ItemSubType.Weapon];
+                    if (SharedModel.AgentEquipmentRankingInfos
+                        .TryGetValue(states.CurrentAvatarKey, out var equipmentRankingMap))
+                    {
+                        var weaponInfo = equipmentRankingMap[ItemSubType.Weapon];
+                        myInfoCell.SetDataAsEquipment(weaponInfo);
+                    }
+                    else
+                    {
+                        myInfoCell.SetEmpty(states.CurrentAvatarState);
+                    }
+
+                    rankScroll.Show(weaponRankingInfos, true);
+                    break;
+                case RankCategory.EquipmentArmor:
+                    if (!isApiLoaded)
+                    {
+                        break;
+                    }
+
+                    var armorRankingInfos = SharedModel.EquipmentRankingInfosMap[ItemSubType.Armor];
```

Comment:
```
 var armorRankingInfos = SharedModel.EquipmentRankingInfosMap[ItemSubType.Armor];
if (SharedModel.AgentEquipmentRankingInfos.TryGetValue(states.CurrentAvatarKey, out equipmentRankingMap))
{
  var armorInfo = equipmentRankingMap[ItemSubType.Armor];
  myInfoCell.SetDataAsEquipment(armorInfo);
}
else
{
  myInfoCell.SetEmpty(states.CurrentAvatarState);
}
```
Too much duplicate code. It looks like it can be handled with a method